lib/ansible/cli/galaxy.py

```diff
@@ -568,10 +568,18 @@ def parse_role_req(requirement):
                 raise AnsibleError("Expecting only 'roles' and/or 'collections' as base keys in the requirements "
                                    "file. Found: %s" % (to_native(", ".join(extra_keys))))
 
-            for role_req in file_requirements.get('roles', []):
```

Comment:
the issue I see and you try to solve is not empty but handle a none value. IMHO this could be simplified with: 
~~~python
for role_req in file_requirements.get('roles') or []:
~~~

Reply:
Make perfect sense. Thanks.